packages/client/src/runtime/getPrismaClient.ts

```diff
@@ -505,12 +513,18 @@ export function getPrismaClient(config: GetPrismaClientConfig) {
     }
 
     getEngine(): Engine {
-      if (this._dataProxy === true) {
+      if (this._dataProxy === true && (TARGET_ENGINE_TYPE === 'data-proxy' || TARGET_ENGINE_TYPE === 'all')) {
```

Comment:
Is esbuild able to tree-shake this now? When I first introduced it, it really had to be `VAR && value` to be able to remove via value injection when it is `false`.

Reply:
Yes, probably they've improved a bit since then. Also, minification removes more dead code then just tree-shaking.